framework/src/wifi_manager/slsi_wifi_utils.c

```diff
@@ -318,12 +346,37 @@ wifi_utils_result_e wifi_utils_scan_ap(void *arg)
 	if (ret != SLSI_STATUS_SUCCESS) {
 		ndbg("[WU] [ERR] Register Scan Callback(%d)\n", ret);
 		return wuret;
-	} 
-	ret = WiFiScanNetwork();
+	}
+	slsi_ap_config_t ap_config;
+	memset((void*)&ap_config, 0, sizeof(slsi_ap_config_t));
+	scan_filter_result.scan_flag = 0;
+	if (arg != NULL) {
+		wifi_utils_ap_config_s *ap_connect_config = (wifi_utils_ap_config_s *)arg;
+		strncpy((FAR char *)ap_config.ssid, (FAR const char *)ap_connect_config->ssid, ap_connect_config->ssid_length);
+		ap_config.ssid_len = ap_connect_config->ssid_length;
+		scan_filter_result.scan_flag = 1;
+		if (scan_filter_result.result_list != NULL) {
+			free_scan_results(scan_filter_result.result_list);
+			scan_filter_result.result_list = NULL;
+		}
+		strncpy((char*)scan_filter_result.scan_ssid, (const char*)ap_config.ssid, ap_connect_config->ssid_length);
+	}
+	ret = WiFiScanNetwork(&ap_config);
 	if (ret != SLSI_STATUS_SUCCESS) {
 		ndbg("[WU] [ERR] WiFi scan fail(%d)\n", ret);
 		return wuret;
-	} 
+	}
+	if (scan_filter_result.scan_flag > 0) {
+		struct timespec abstime;
+		(void)clock_gettime(CLOCK_REALTIME, &abstime);
+		abstime.tv_sec += (SLSI_SCAN_INTERVAL + 1);
```

Comment:
I assume this semaphore wait timeout is used for scan timeout.
SLSI_SCAN_INTERVAL seems to be interval between each scan operation in slsi WiFi low-level driver.
not time-out specified by driver.
so my question is why use this macro here?

Reply:
The scan should be done within SLSI_SCAN_INTERVAL it is the time of next scan start. If the scan done event is not reached when the next scan should be started, there must be something wrong and the scan fails.